coops-ui/tests/unit/Directors.spec.ts

```diff
@@ -168,110 +137,89 @@ describe('Directors.vue', () => {
     expect(directorListUI[1].innerHTML).toContain('<span>Cease</span>')
   })
 
-  it('buttons/actions are disabled when the AGM date has not been set', () => {
-    // clear AGM Date in AGMDate component
-    childvm.$store.state.agmDate = null
+  it('disables buttons/actions when AGM Date is invalid', done => {
+    // invalidate AGM Date
+    vm.$store.state.agmDateValid = false
 
     Vue.nextTick(() => {
-      var directorListUI = childvm.$el.querySelectorAll('.director-list .container')
-
       // confirm that flag is set correctly
-      expect(childvm.agmEntered).toEqual(false)
+      expect(vm.agmDateValid).toEqual(false)
+
+      const directorListUI = vm.$el.querySelectorAll('.director-list .container')
 
       // check that buttons are disabled (checks first button in first director, plus the Add New Director button)
-      expect(
-        directorListUI[0].querySelector('.cease-btn').attributes.getNamedItem('disabled').value
-      ).toEqual('disabled')
-      expect(
-        childvm.$el.querySelector('.new-director-btn').attributes.getNamedItem('disabled').value
-      ).toEqual('disabled')
+      expect(directorListUI[0].querySelector('.cease-btn').disabled).toBe(true)
+      expect(vm.$el.querySelector('.new-director-btn').disabled).toBe(true)
+
+      done()
     })
   })
 
-  it('buttons/actions are enabled when the AGM date has been set', () => {
-    var directorListUI = childvm.$el.querySelectorAll('.director-list .container')
-
-    // set AGM Date in AGMDate component
-    childvm.$store.state.agmDate = '2018-06-18'
+  it('enables buttons/actions when AGM Date is valid', done => {
+    // validate AGM Date
+    vm.$store.state.agmDateValid = true
 
     Vue.nextTick(() => {
       // confirm that flag is set correctly
-      expect(childvm.agmEntered).toEqual(true)
-
-      // check that buttons are enabled (checks first button in first director, plus the Add New Director button)
-      expect(
-        directorListUI[0].querySelector('.cease-btn').attributes.getNamedItem('disabled')
-      ).toBeNull()
-      expect(
-        childvm.$el.querySelector('.new-director-btn').attributes.getNamedItem('disabled')
-      ).toBeNull()
-    })
-  })
+      expect(vm.agmDateValid).toEqual(true)
 
-  it('buttons/actions are enabled when "No AGM" option set', () => {
-    var directorListUI = childvm.$el.querySelectorAll('.director-list .container')
-
-    // set "No AGM" in AGMDate component
-    childvm.$store.state.noAGM = true
-
-    setTimeout(() => {
-      // confirm that flag is set correctly
-      expect(childvm.agmEntered).toEqual(true)
+      const directorListUI = vm.$el.querySelectorAll('.director-list .container')
```

Comment:
Is there a reason this was switched from var to const? If it serves the same purpose I'm resistant to making the change, as well as some of the other changes that may not functionally change anything (rewrite of click() function for example) because of the overlap with other team members.

Reply:
In TypeScript we should be using `const` where possible, and `let` if it's a basic type that can change its value (that is, not for objects unless you need to change the reference). The JavaScript `var` and its weird scoping should be avoided, unless the weird scoping is really needed - and I think that's a code smell.

Reply:
It's stronger typing (esp since we're using TS, esp since a TS linter will complain about using var when you should be using const or let instead).